### PR TITLE
Mark --snapshot-count as deprecated in v3.6 and will be decommissioned in 3.7

### PR DESCRIPTION
**Pull request by ahrtr**
Resolve https://github.com/etcd-io/etcd/issues/18995

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

cc @fuweid @serathius

### CODE REVIEW

**Comment by ahrtr** [comment]
cc @ivanvc and @jmhbnz

**Review by fuweid — APPROVED** [review]
LGTM

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @ahrtr

**Review by ivanvc — COMMENTED** [review]
LGTM. Thanks, @ahrtr.

**Review by ivanvc — APPROVED** [review]

**Review by serathius — APPROVED** [review]

File: server/etcdmain/config.go [hunk 1]
```diff
@@ -165,6 +170,20 @@ func (cfg *config) parse(arguments []string) error {
...
+			warningsForDeprecatedFlags = append(warningsForDeprecatedFlags, msg)
+		}
+	})
+	if len(warningsForDeprecatedFlags) > 0 {
+		if lg := cfg.ec.GetLogger(); lg != nil {
+			for _, msg := range warningsForDeprecatedFlags {
+				lg.Warn(msg)
+			}
+		}
+	}
```
Comment (ahrtr) [inline server/etcdmain/config.go:185]: It seems that it only checks the command line flags. We should check the config file input as well. Proposed actions:
- Move this code snip into `configFromCmdLine`;
- add similar check in `configFromFile`

**Pull request merged**